framework/auth/views.py

```diff
@@ -815,62 +815,64 @@ def send_confirm_email(user, email, renew=False, external_id_provider=None, exte
     :return:
     :raises: KeyError if user does not have a confirmation token for the given email.
     """
+    def send_email_task():
```

Comment:
Wondering why define this `send_email_task()` inside the `send_confirm_email()`?

Usually, we define `send_confirm_email_async()` that calls `send_confirm_email()` with arguments properly passed. Then, need to refactor places where `send_confirm_email()` is called to call `send_confirm_email_async()`. Finally, need to update unit tests (if any).